tests/queries/0_stateless/02391_recursive_buffer.sql

```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS test;
+CREATE TABLE test (key UInt32) Engine = Buffer(currentDatabase(), test, 16, 10, 100, 10000, 1000000, 10000000, 100000000);
+SELECT * FROM test; -- { serverError 269 }
+SELECT * FROM system.tables WHERE table = 'test' AND database = currentDatabase() FORMAT Null; -- { serverError 269 }
```

Comment:
Note, that concurrent run this test with `select * from system.tables` queries (in some other tests) may produce `INFINITE_LOOP` errors there, but maybe this test should be run in parallel, and instead tests that do `select * from system.tables` should not.